### PR TITLE
Add Distance API support. Fixes #36

### LINKED ISSUE #36: Support distances api

### PR DESCRIPTION
**Pull request by tmcw**
cc @freenerd for the review

### CODE REVIEW

File: lib/services/distance.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+var MapboxDistance = makeService('MapboxDistance');
+
+/**
+ * Compute a distance table of all paths between a set of waypoints.
+ * Consult the [Mapbox Distance API](https://www.mapbox.com/developers/api/distance/)
+ * for more documentation.
+ *
+ * @param {Array<Array<number>>} waypoints an array of objects with `latitude`
+ * and `longitude` properties that represent waypoints in order. Up to
```
Comment (freenerd) [inline lib/services/distance.js]: it's `[ longitude, latitude ]`

File: lib/services/distance.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+var invariant = require('invariant'),
+  makeService = require('../make_service'),
+  constants = require('../constants');
+
+var MapboxDistance = makeService('MapboxDistance');
+
+/**
+ * Compute a distance table of all paths between a set of waypoints.
```
Comment (freenerd) [inline lib/services/distance.js]: note that the distance table returns travel times and not actual distances?

File: lib/services/distance.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+ *   console.log(results);
+ * });
+ *
+ * // Results is an object like:
+ * { durations:
+ *   [ [ 0, 1196, 3977, 3415, 5196 ],
+ *     [ 1207, 0, 3775, 3213, 4993 ],
+ *     [ 3976, 3774, 0, 2650, 2579 ],
+ *     [ 3415, 3212, 2650, 0, 3869 ],
+ *     [ 5208, 5006, 2579, 3882, 0 ] ] }
```
Comment (freenerd) [inline lib/services/distance.js]: for the effect of demonstration include a `null` value here?

**Comment by freenerd** [comment]
It the api returns a non-200 code, will that show up in the `err` param of the callback? The API returns a fair share of 40X for invalid accessToken, invalid coordinates etc with a descript body.

File: lib/services/distance.js (hunk 3 above)
Comment (tmcw) [inline lib/services/distance.js]: :+1: added a null example

**Comment by tmcw** [comment]
Any JSON error responses returned by the server will be available as the `err` argument to the callback

**Pull request merged**